src/posix/platform/trel.cpp

```diff
@@ -0,0 +1,547 @@
+    // number or new IPv6 address), or when the peer is removed.
+}
+
+OT_TOOL_WEAK void trelDnssdStopBrowse(void)
+{
+    // This function stops the ongoing DNS-SD browse started from an
+    // earlier call to `trelDnssdStartBrowse()`.
+}
+
+OT_TOOL_WEAK void trelDnssdRegisterService(uint16_t aPort, const uint8_t *aTxtData, uint8_t aTxtLength)
+{
+    // This function registers a new service to be advertised using
+    // DNS-SD.
+    //
+    // The service name is "_trel._udp". The platform should use its own
+    // hostname, which when combined with the service name and the
+    // local DNS-SD domain name will produce the full service instance
+    // name, for example "example-host._trel._udp.local.".
+    //
+    // The domain under which the service instance name appears will
+    // be 'local' for mDNS, and will be whatever domain is used for
+    // service registration in the case of a non-mDNS local DNS-SD
+    // service.
+    //
+    // A subsequent call to this function updates the previous service.
+    // It is used to update the TXT record data and/or the port
+    // number.
+    //
+    // The `aTxtData` buffer is not persisted after the return from this
+    // function. The platform layer MUST not keep the pointer and
+    // instead copy the content if needed.
+
+    OT_UNUSED_VARIABLE(aPort);
+    OT_UNUSED_VARIABLE(aTxtData);
+    OT_UNUSED_VARIABLE(aTxtLength);
+}
+
+OT_TOOL_WEAK void trelDnssdRemoveService(void)
+{
+    // This function removes any previously registered "_trel._udp"
+    // service using `platTrelRegisterService()`. Device must stop
+    // advertising TREL service after this call.
+}
+
+OT_TOOL_WEAK void trelDnssdUpdateFdSet(fd_set *aReadFdSet, fd_set *aWriteFdSet, int *aMaxFd, struct timeval *aTimeout)
+{
+    // This function can be used to update the file descriptor sets
+    // by DNS-SD layer (if needed).
+
+    OT_UNUSED_VARIABLE(aReadFdSet);
+    OT_UNUSED_VARIABLE(aWriteFdSet);
+    OT_UNUSED_VARIABLE(aMaxFd);
+    OT_UNUSED_VARIABLE(aTimeout);
+}
+
+OT_TOOL_WEAK void trelDnssdProcess(otInstance *aInstance, const fd_set *aReadFdSet, const fd_set *aWriteFdSet)
+{
+    // This function performs processing by DNS-SD (if needed).
+
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aReadFdSet);
+    OT_UNUSED_VARIABLE(aWriteFdSet);
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+// otPlatTrel
+
+void otPlatTrelEnable(otInstance *aInstance, uint16_t *aUdpPort)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    assert(sInitialized);
+
+    VerifyOrExit(!sEnabled);
+
+    PrepareSocket(*aUdpPort);
+    trelDnssdStartBrowse();
+
+    sEnabled = true;
+
+exit:
+    return;
+}
+
+void otPlatTrelDisable(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    assert(sInitialized);
+    VerifyOrExit(sEnabled);
+
+    close(sSocket);
+    sSocket = -1;
+    trelDnssdStopBrowse();
+    trelDnssdRemoveService();
```

Comment:
```suggestion
    trelDnssdRemoveService();
    
    sEnabled = false;
```
seems forget to set `sEnabled` to false?